### DISCUSSION TITLE
How do I set Brush.Color and Pen.Color parameters of a TShape?

### DISCUSSION THREAD

**Discussion by IanC216** [Embarcadero/DelphiVCL4Python#56] [Q&A]
Hi, I'm loving DelphiVCL so far but I have a slight problem.
I have a Shape object on my form... 

import delphivcl as vcl

  self.Shape1 = vcl.Shape(self)
  self.Shape1.SetProps(Parent=self, Left=216, Top=136, Height=80, Width=105, Shape='stRoundSquare')

... This appears on the form but only with a white background and a black pen. I can't find the relevant kwargs to set the Brush.Color, Pen.Color or Pen.Width properties. How do I do this please? I've tried 'Brush.Color=', 'Brush_Color=', 'BrushColor=', 'Brush[Color]=' and quite a few more to no avail.

While I'm here, I can't change VCL style either. The demo 'activityindicator.py' file crashes when I try to change style. If you could fix that bug, I'll be eternally grateful.

Thanks.

**Comment by IanC216** [comment]
Never mind. I found out.
    self.Shape1.Brush.SetProps(Color=...

The VCL style problem remains, however.

**Reply by lmbelo** [reply]
Hello @IanC216,
@Priyatham10 and I will review the "activityindicator.py" script. Can you open an issue, please?

**Reply by IanC216** [reply]
Hi, @lmbelo,
I didn't raise it as an issue because the problem I'm having with the "activityindicator.py" script has already been raised as an issue. Issue #50 raised by @fansxs on March 18th. Should I raise another identical issue?
I have the exact same problem and nothing I've tried (so far) works. Firstly, "TypError: cannot unpack non-iterable bool object" is raised at the line "valid_style and not (style_info.name in self.cbx_vcl_style.Items):" when I try to change style.
Secondly, when the above line is edited (just as fansxs did) I get "Typerrror: Call 'LoadFromFile' returned a value that could not be converted to Python"

I should add, I'm using version DelphiVCL 1.0.3

**Reply by IanC216** [reply]
More information in the code comments:
def __on_load_style_click(self, sender):
    if self.od_styles.Execute():
        for file in os.listdir(self.od_styles.FileName):
            if file.endswith(".vsf"):
                style_file_path = os.path.join(self.od_styles.FileName, file)
                print(style_file_path) # EDIT prints first .vsf file in the folder (Amakrits.vsf)
                valid_style = self.__sm.IsValidStyle(style_file_path) 
                print(valid_style) # EDIT prints 'True'
                if valid_style:  # and not (style_info.Name in self.cbx_vcl_style.Items): # EDIT
                    self.__sm.LoadFromFile(style_file_path)  # ERROR HERE - TypeError: ... value that could not be converted to Python
